codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FlowController.java

```diff
@@ -22,6 +22,12 @@
 public interface Http2FlowController {
 
     /**
+     * Gets the {@link Http2Connection.PropertyKey} used for accessing the {@link Http2Stream.FlowControlState} property
+     * that this controller associates with each {@link Http2Stream}.
+     */
+    Http2Connection.PropertyKey stateKey();
```

Comment:
In summary I think we could get this PR in and then re-evaluate if any improvements need to be made :)